src/net.cpp

```diff
@@ -2505,7 +2507,7 @@ bool CConnman::InitBinds(const Options& options)
         fBound |= Bind(CService(inaddr6_any, GetListenPort()), BF_NONE, NetPermissionFlags::None);
         fBound |= Bind(CService(inaddr_any, GetListenPort()), !fBound ? BF_REPORT_ERROR : BF_NONE, NetPermissionFlags::None);
     }
-    return fBound;
+    return fBound && (bound_onion || options.onion_binds.empty());
```

Comment:
I dislike making this logic more and more complex, special-casing for onion binds. I'd prefer something more general instead that will solve this and surrounding issues.

Reply:
Changed the code to consider bind failures fatal.